chapters/en/slides/module7/module7_04.Rmd

```diff
@@ -0,0 +1,342 @@
+All the buttons, banners, etc that you see when you go to a web page
+are not images but HTML code
+that your browser can convert into a graphical elements such as buttons.
+
+Vega-Lite code is rendered in a similar way.
+Modern browsers know how to draw graphical elements from the Vega-Lite code,
+so we never need to create any images.
+
+So what does this Vega-Lite code look like?
+
+---
+
+## The Vega-Lite JSON specification for charts
+
+```python
+print(chart.to_json())
+```
+
+```json
+{
+  "config": {
+    "view": {
+      "continuousHeight": 300,
+      "continuousWidth": 400
+    }
+  },
+  "data": {
+    "url": "https://vega.github.io/vega-datasets/data/cars.json"
+  },
+  "encoding": {
+    "x": {
+      "field": "Horsepower",
+      "type": "quantitative"
+    },
+    "y": {
+      "field": "Miles_per_Gallon",
+      "type": "quantitative"
+    }
+  },
+  "mark": "circle"
+}
+```
+
+
+[Open the Chart in the Vega Editor](https://vega.github.io/editor/#/url/vega-lite/N4IgJAzgxgFgpgWwIYgFwhgF0wBwqgegIDc4BzJAOjIEtMYBXAI0poHsDp5kTykBaADZ04JACyUAHJQCMlAFYQ2AOxAAaEFBUAzGmTShiNOAHcDmlZhrKGbBhAAScPVjQBmAAweNW5VZt2EADqNAAm9GhiXgC+0RqhSJgoqKAMAE6CaBjYeIS8FNR0jCzs+QIJSRBwmBAEFUgEUEhpEApKqnEgcMpaodb6KSAAHua6cIKhWQ5sLXA4bCZwaeogmACeOHBZAI4MSH50iTSkIJ1ro8YTWQCyNIJwEAD6m2mPAOJIgoIqK+ubO3sDkkrCdYhpkGkANZZKA0NJQe6nIA)
+
+Notes:
+Instead of creating a new programming language to store its code,
+Vega-Lite uses the established JSON file format.
+
+JSON is already widely used on the web as a storage format for text data,
+so this approach means that Vega-Lite does not need to reinvent the wheel.
+Instead, there are rules for which words are allows to appear in the JSON
+file for it to be valid Vega-Lite code,
+which is referred to as the "specification" (or "spec" for short).
+
+You can see an example of how this looks for an Altair visualization
+by using the `to_json` method as we have done in this slide.
+You can see that the words are the same we use in Altair,
+but the format of the JSON specification
+is a bit more verbose than  the Altair syntax.
+
+You can paste this JSON directly into Vega-Lite online editor
+and you will see that it renders correctly.
+We included a link for this in the slide that you can try.
+
+This is the same that happens if we click the "Open in Vega Editor" entry
+in the three dot action button that is automatically included in each Altair chart.
+
+If you look at the HTML file we saved in the previous step
+you will see that this JSON spec is included together with some additional code
+that tells your browser to use Vega-Lite's web libraries to render this JSON spec.
+
+---
+
+## Charts can be rendered offline
+
+
+- [Link to install the offline Vega-Lite extension in JupyterLab](https://altair-viz.github.io/user_guide/display_frontends.html#displaying-in-jupyterlab)
+- Code to include a fallback image representation of your chart for offline rendering:
+
+    ```
+    alt.renderers.enable('mimetype')
+    ```
+
```

Comment:
Some sort of graphic is needed for this slide...

Reply:
I don't know what graphic to include since this doesn't change anything appearance-wise, but I included an example of what the text blob looks like in the notebook.